### ISSUE TITLE
Why dont we use non_max_suppression in training loop?

### ISSUE THREAD

**Issue by QueeneTam** [eriklindernoren/PyTorch-YOLOv3#583]
It seems that non_max_suppression is only being used in inference stage not in training stage. 
Why dont we use non_max_suppression in training loop? Is there any reasons?

**Comment by crownz-sec** [comment]
Training stage is used to let the network learn the labels of images, just like other training step,every step's result is not accurate.So there are many wrong bounding box at the end of every mini-batch,and you don't need to use non\_max\_suppression to adjust them.

**Comment by Flova** [comment]
I agree with @crownz-sec. I close it for now but feel free to reopen this issue if a further discussion is needed.
